### DISCUSSION TITLE
Memory Leak on "bluealsa" process

### DISCUSSION THREAD

**Discussion by daniele86rm** [Arkq/bluez-alsa#624] [General]
Hello Guys,
I'm testing "bluealsa" project on my embedded application and I have the following issue:
my bluetooth device (bluetooth headset) is already paired and trusted on "bluetoothd" daemon, when I power-on the bluetooth headset, after the automatically bluetooth connection, I see that the "bluealsa" memory increase and is never released, even when the bluetooth device is disconnected.
Each new connection increase bluealsa memory as shown in the attached figure.
If I restart the "bluealsa" process the memory is released.
Is there some advice on this issue? Thanks in advance.

![bluealsaMemoryUsage](https://user-images.githubusercontent.com/56957776/229797032-e82068fe-491d-4681-8daa-604d27fcca3b.jpg)

**Comment by arkq** [comment]
> Each new connection increase bluealsa memory as shown in the attached figure.

Are you connecting and disconnecting the same device over and over again, or you've got more than one device? There are indeed some internal structures which are not release on disconnection (maybe you are seeing this thing), however, memory is allocated per-device and it's not a leak - memory is released when HCI interface is removed from the system (you can test it by restarting `bleutoothd`). Or maybe there is indeed a leak somewhere... I'll try to check that. Which version of bluez-alsa you are using (hopefully fresh build from master branch :D)?

**Comment by daniele86rm** [comment]
Yes, I'm connecting and disconnecting the same device but it seems that the memory is not released. I have only one bluetooth device.  I have tried to restart bluetoothd process but the memory occupied from bluealsa is not released. The only way to release bluealsa memory is to restart bluealsa service. I'm using bluealsa "rev335f48c-231019" version. Now I will try to build the master branch version of bluez-alsa and I let you know. Thanks a lot.

**Comment by daniele86rm** [comment]
I'm testing the latest version of bluez-alsa and I confirm that the memory not increase when I connecting/disconnecting bluetooth device. I have noticed that in the last version is needed to spiecified bluetoot profile to start "bluealsa" process (exapmle /usr/bin/bluealsa -p hfp-ag), meanwhile in the old versione is not requested.  You can confirm it?
Thanks for your support.

**Reply by arkq** [reply]
> I have noticed that in the last version is needed to spiecified bluetoot profile to start "bluealsa" process (exapmle /usr/bin/bluealsa -p hfp-ag), meanwhile in the old versione is not requested. You can confirm it?

Yes, that's the new behavior. There is no built-in default any more. The same thing (almost) is for enabled codecs. Only basic (covered by the HFP/A2DP specs) codecs are enabled by default. Any vendor codec has to be enabled on command line with `-c`, e.g. `-c aptx`.
